app/controllers/api/v1/targets_controller.rb

```diff
@@ -31,6 +47,22 @@ def target_params
       def target
         @target = Target.find(params[:id])
       end
+
+      def search_match(radius, latitude, longitude, topic_id)
+        Target.within(
+          radius,
+          :units => :miles,
+          :origin => [ latitude, longitude ]
+        ).where(topic_id: topic_id).where.not(user_id: current_user.id)
+      end
+
+      def set_conversation(user_a, user_b)
+        if Conversation.between(user_a, user_b).present?
```

Comment:
Maybe you can use `first_or_create`

https://apidock.com/rails/ActiveRecord/Relation/first_or_create